systemFlake.nix

```diff
@@ -1,6 +1,66 @@
-{ flake-utils-plus }:
+{ flake-utils-plus }: let
 
-{ self
+ builtinPackagesBuilder = pkgs: channels:
```

Comment:
In my opinion this function is complex/large enough to deserve its own file. But more importantly I think you should export it in flake-utils. It has a lot of value for others. I'm indifferent on whether this should be done by default. 

But I think this would be better done as a function exported by fup. And then its agreed to do it by default, then you can do `packagesBuilder ? flake-utils-plus.builtinPackagesBuilder`.

Although in this case perhaps it should be named differently. overlaysPackagesBuilder? autoPackagesBuilder?

Reply:
Good points. This is only my first iteration, and I definitly have half the worl still before me. :wink:

Reply:
If it is getting moved to a separate file - you probably will need stuff like `optionals` and `mkIf` - probably a utils file should be created while we are at it. 

I'm fine with exposing publically via `self.lib.internal.<function-name>` or having that reference privately, it up to you guys to decide

Reply:
> If it is getting moved to a separate file - you probably will need stuff like optionals and mkIf - probably a utils file should be created while we are at it.

I actually think it's a nice coding style of upstream flake-utils that each file can be entirely read by its own, without external references. While it _might_ duplicate a little code, it's much more polite to first time readers of the fup repo, and transcending this, it's probably good marketing for this repo.